tools/importer/rules/marquee.js

```diff
@@ -1,142 +1,233 @@
-    const cells = [[marqueeVariation(blockHeight)]];
-    cells.push([bgcolor]);
-    cells.push([children[1], children[0]]);
-    const table = WebImporter.DOMUtils.createTable(cells, document);
-    block.before(document.createElement('hr'));
-    table.classList.add('import-table');
-    block.replaceWith(table);
-  }
-  return anotherVersionFlag;
-};
+function createMarqueeHeader({ block }) {
+  const blockHeight = parseInt(block?.getAttribute('data-height') ?? 0, 10);
+  const marqueeHeader = marqueeVariation(blockHeight);
+  return [[marqueeHeader]];
+}
 
-export default function createMarqueeBlocks(block, document) {
-  if (anotherVersionMarquee(block, document)) {
-    return;
+/**
+ *
+ * @param {{block: HTMLDivElement}} inputParams
+ * @returns {Array<String | HTMLAnchorElement | HTMLImageElement}
+ */
+function createMarqueeBackground({ block, document }) {
+  const isVideo = !!block?.querySelector('.video-Wrapper source');
+  const isBackgroundColor = !!block?.querySelector('div[data-bgcolor]');
+  const isBgImage = !!block.querySelector('div[style]');
+  if (isVideo) {
+    const video = makeVideo(block, document);
+    return [[video]];
   }
-  const blockHeight = parseInt(block.getAttribute('data-height'), 10);
-  const textElement = document.createElement('div');
-
-  // Create table for marquee
-  const cells = [[marqueeVariation(blockHeight)]];
-
-  // Check if marquee has background video
-  const videoWrapper = block.querySelector('.video-Wrapper source');
-  if (videoWrapper) {
-    const backgroundVideoURL = videoWrapper.getAttribute('src');
-    const a = document.createElement('a');
-    a.innerHTML = backgroundVideoURL;
-    a.setAttribute('href', backgroundVideoURL);
-    cells.push([a]);
-  } else {
-    const bgImage = block.querySelector('div[style]');
-    if (bgImage) {
-      const image = bgImage?.getAttribute('style').split('"')[1];
-      let imageUrl = [];
-      if (image.indexOf('http://localhost:3001') !== -1) {
-        imageUrl = image.split('http://localhost:3001');
-      }
-      const imageTag = document.createElement('img');
-      imageTag.src = imageUrl.length ? `https://www.adobe.com${imageUrl[1]}` : image;
-      cells.push([imageTag]);
-    }
+  if (isBackgroundColor) {
+    const bgcolor = makeBGColor(block);
+    return [[bgcolor]];
+  }
+  if (isBgImage) {
+    const bgImage = makeBGImage(block, document);
+    return [[bgImage]];
   }
+  return [];
+}
 
-  // Check if marquee has icon
-  const images = block.querySelectorAll('img');
-  images.forEach((image) => {
-    let imgSrc = image.getAttribute('src');
-    if (imgSrc && (imgSrc.indexOf('.svg') + 1)) {
-      if (imgSrc.indexOf('https') === -1) {
-        imgSrc = `https://www.adobe.com/${imgSrc}`;
-      }
-      const imgLink = document.createElement('a');
-      imgLink.innerHTML = imgSrc;
-      imgLink.setAttribute('href', imgSrc);
-      textElement.appendChild(imgLink);
-    }
+/**
+ * Function for building Marquee,
```

Comment:
JsDoc is not attached to the right function

Reply:
done